crates/protogen/src/metastore/types/catalog.rs

```diff
@@ -494,47 +494,38 @@ impl From<FunctionType> for catalog::function_entry::FunctionType {
         }
     }
 }
+
+/// The runtime preference for a function.
+/// It is important that this MUST match the values and order in `protogen::gen::metastore::catalog::function_entry::RuntimePreference`
+#[repr(i32)]
 #[derive(Debug, Clone, Copy, Arbitrary, PartialEq, Eq)]
 pub enum RuntimePreference {
-    Unspecified,
-    Local,
-    Remote,
+    Unspecified = 0,
+    Local = 1,
+    Remote = 2,
 }
 
 impl TryFrom<i32> for RuntimePreference {
     type Error = ProtoConvError;
     fn try_from(value: i32) -> Result<Self, Self::Error> {
-        catalog::function_entry::RuntimePreference::from_i32(value)
-            .ok_or(ProtoConvError::UnknownEnumVariant(
-                "RuntimePreference",
-                value,
-            ))
-            .and_then(|t| t.try_into())
+        let pref = catalog::function_entry::RuntimePreference::from_i32(value).ok_or(
+            ProtoConvError::UnknownEnumVariant("RuntimePreference", value),
+        )?;
+        Ok(pref.into())
     }
 }
 
-impl TryFrom<catalog::function_entry::RuntimePreference> for RuntimePreference {
-    type Error = ProtoConvError;
-    fn try_from(value: catalog::function_entry::RuntimePreference) -> Result<Self, Self::Error> {
-        Ok(match value {
-            catalog::function_entry::RuntimePreference::Unspecified => {
-                RuntimePreference::Unspecified
-            }
-            catalog::function_entry::RuntimePreference::Local => RuntimePreference::Local,
-            catalog::function_entry::RuntimePreference::Remote => RuntimePreference::Remote,
-        })
+impl From<catalog::function_entry::RuntimePreference> for RuntimePreference {
+    fn from(value: catalog::function_entry::RuntimePreference) -> Self {
+        // This is safe because the enum values should be identical.
+        // If they're not, then we need to update the proto repr
+        unsafe { std::mem::transmute(value) }
```

Comment:
I'd prefer sticking the the match to be more explicit since this is a write once thing. Also the match would actually compile down to nothing.